python/tank/platform/import_stack.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2013 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Manages a thread-safe stack of Toolkit imports.
+"""
+
+import threading
+
+
+class ImportStack(object):
+    """
+    Manages a thread-safe stack of Toolkit imports.
```

Comment:
Maybe add something like "used in conjunction with the `_get_current_bundle()` method - for more details, see `platform/__init__.py`